src/item_factory.cpp

```diff
@@ -5469,6 +5470,27 @@ std::vector<const itype *> Item_factory::find( const std::function<bool( const i
     return res;
 }
 
+const std::vector<const itype *> &Item_factory::find_holster_for( const itype &it )
+{
+    const itype_id iid = it.get_id();
+    if( type_contained_in_holsters.count( iid ) > 0 ) {
+        return type_contained_in_holsters.at( iid );
+    }
+    const item itm( &it );
+
+    type_contained_in_holsters[iid] = Item_factory::find( [&itm]( const itype & e ) {
+        if( !e.can_use( "holster" ) ) {
+            return false;
+        }
+        const holster_actor *ptr = dynamic_cast<const holster_actor *>
+                                   ( e.get_use( "holster" )->get_actor_ptr() );
+        const item holster_item( &e );
+        return ptr->can_holster( holster_item, itm ) && !item_is_blacklisted( holster_item.typeId() );
```

Comment:
Is the cache updated on blacklist changes?

Reply:
It is not. I guess we don't need this at all. Thanks for bringing this up whether it concludes to be not needed or completely wrong.

Reply:
I removed the check.

Reply:
Given the check exists in the old code, I would be wary that removing it removes functionality. That said, I don't really use blacklisting when I play, so I don't know what it would cause.

Reply:
I think we could refresh the cache on blacklistings... If that is wanted.

Reply:
I kept the blacklists in the new PR #80203.